gammapy/datasets/metadata.py

```diff
@@ -0,0 +1,157 @@
+import logging
+from typing import Optional, Union
+import numpy as np
+from astropy.coordinates import SkyCoord
+from pydantic import ValidationError, validator
+from gammapy.utils.metadata import CreatorMetaData, MetaData
+
+__all__ = ["MapDatasetMetaData"]
+
+
+class MapDatasetMetaData(MetaData):
```

Comment:
Should we not create a class for the Dataset object? and having daughter classes with their specificities? This would avoid code duplication...

Reply:
Or a more general `ObjectMetaData` as @registerrier suggested